src/modules/socket.ts

```diff
@@ -2,147 +2,225 @@ import { io, Socket } from "socket.io-client";
+                                new CustomEvent("reset")
+                            );
+                        }
 
-  sendMessage = (message: string) => {
-    this.checkForErrors();
-    this.socket.emit(SocketEventType.MESSAGE, message, (response: string) => {
-      console.log("Response: " + response);
-    });
-  };
+                        console.log(this.video);
+                        const playCode = `const videoPlayer = netflix.appContext.state.playerApp.getAPI().videoPlayer;
+                        const player = videoPlayer.getVideoPlayerBySessionId(videoPlayer.getAllPlayerSessionIds()[0]);
+                        player.play();`;
+                        document.documentElement.setAttribute(
+                            "onreset",
+                            playCode
+                        );
+                        document.documentElement.dispatchEvent(
+                            new CustomEvent("reset")
+                        );
+                        console.log("Video is played!");
+                        break;
+                    case "pause":
+                        console.log(this.video);
+                        const pauseCode = `const videoPlayer = netflix.appContext.state.playerApp.getAPI().videoPlayer;
+                        const player = videoPlayer.getVideoPlayerBySessionId(videoPlayer.getAllPlayerSessionIds()[0]);
+                        player.pause();`;
+                        document.documentElement.setAttribute(
+                            "onreset",
+                            pauseCode
+                        );
+                        document.documentElement.dispatchEvent(
+                            new CustomEvent("reset")
+                        );
+                        console.log("Video is paused!");
+                        break;
+                    case "seeked":
+                        console.log("seeeek", this.seek);
+                        const time = message.currentTime * 1000;
+                        const seekCode = `const videoPlayer = netflix.appContext.state.playerApp.getAPI().videoPlayer;
+                            const player = videoPlayer.getVideoPlayerBySessionId(videoPlayer.getAllPlayerSessionIds()[0]);
+                            player.seek(${time});`;
+                        document.documentElement.setAttribute(
+                            "onreset",
+                            seekCode
+                        );
+                        document.documentElement.dispatchEvent(
+                            new CustomEvent("reset")
+                        );
+                        document.documentElement.removeAttribute("onreset");
+                        console.log("Video is seeked!", message.currentTime);
+                        this.messageEvent.seek = true;
+                        break;
+                    
+                }
+                setTimeout(() => {
```

Comment:
This is the semaphore which is responsible for blocking event sending when receiving the event, it is very important because without it users in the room would start a loop of play and pause after a seek